test/ARCHITECTURE.md

```diff
@@ -0,0 +1,267 @@
+    message?"}
+    retry["Retry three times
+    to be robust against
+    test failures"]
+    testfailed["Test failed"]
+    failure_policy{"Known issue?"}
+    
+    finished --> succeeded
+    succeeded --> |Yes| skipped
+    succeeded --> |No| todonosuccess
+    
+    skipped --> |Yes| skiptest[Show as skipped test]
+    skipped --> |No| affected
+    
+    affected --> |Yes| affecteddone
+    affected --> |No| todosuccess
+    
+    todosuccess --> |Yes| todosucceeded
+    todosuccess --> |No| done[Test succeeded]
+    
+    todonosuccess --> |Yes| todofail
+    todonosuccess --> |No| failure_policy
+    
+    failure_policy --> |Yes| known_issue[Show as known issue]
+    failure_policy --> |No| pixel_journal_failure
+    
+    pixel_journal_failure --> |Yes| testfailed
+    pixel_journal_failure --> |No| retry
+```
+
+* `Skipped` -  tests can be skipped because the test can not run on the given `TEST_OS`.
+* `Affected` - tests to be retried to make sure any changes affecting them do not lead to flaky tests.
+* `Todo` - tests which are incomplete and expected to fail.
+* `Known issue` - Naughties are expected test failures due to known issues in software we test,
+                  we still run the test but if the test error output matches a known naughty it
+                  is skipped. The [bots](github.com/cockpit-project/bots) repository keeps track
+                  of all our known naughties per distro. (The bots repository has automation
+                  setup to see if a naughty is still affected and if not open a pull request to
+                   drop it).
+* `Failed` - tests can fail due to our test shared infrastructure, instead of
+             letting the whole test run fail, we re-try them unless `--no-retry-fail` is
+             passed.
+
+### Continuous Integration (CI)
+
+In CI we have two entry points, one for our tests which runs on our own
+managed infrastructure by [cockpituous](https://github.com/cockpit-project/cockpituous/)
+and one for tests which run on the [testing farm (TF)](https://docs.testing-farm.io/).
+
+For our own managed infrastructure the entry point of the Cockpit tests is
+`test/run` this bash script expects a `TEST_OS` environment variable to be set
+to determine what distribution to run the tests under and a `TEST_SCENARIO`
+environment variable to determine the type of test. Currently we support these
+different scenario's:
+
+* `devel` - runs tests with coverage enabled and generates a html file with
+  coverage information
+* `pybridge` - runs tests with the Python bridge (soon to be deprecated after all
+             images are moved over to the Python bridge)
+* `firefox` - runs tests using the Firefox browser instead of Chrome
+* `networking` - runs all networking related tests
+* `storage` - runs all storage related tests
+* `expensive` - runs all expensive tests (usually tests which reboot/generate a new initramfs)
+* `other` - runs all non-networking/storage/expensive tests.
```

Comment:
I feel like this is going to become quickly out of sync with test/run and the bots testmap. Please either move that to test/run and link to it from this documentation, or add a reminder comment to test/run to also update this documentation on scenario changes. TBH I prefer the former -- I feel like this document already has too much detail which is prone to getting out of sync with the code.